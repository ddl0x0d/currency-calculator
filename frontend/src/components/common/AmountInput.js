import Form from 'react-bootstrap/Form';
import PropTypes from 'prop-types';
import React from 'react';

const propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  step: PropTypes.number,
  handleChange: PropTypes.func.isRequired
};

const defaultProps = {
  step: 1
};

const AmountInput = props => {
  return (
    <Form.Group>
      <Form.Label className="small mb-0">{props.label}</Form.Label>
      <Form.Control
        id={props.id}
        type="number"
        size="sm"
        value={props.value}
        min={0}
        step={props.step}
        onChange={e => props.handleChange(parseFloat(e.target.value))}
      />
    </Form.Group>
  );
};

AmountInput.propTypes = propTypes;
AmountInput.defaultProps = defaultProps;

export default AmountInput;
