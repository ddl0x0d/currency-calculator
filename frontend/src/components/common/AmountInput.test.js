import AmountInput from './AmountInput';
import { mount, shallow } from 'enzyme';
import React from 'react';

describe('<AmountInput />', () => {
  const handleChange = jest.fn().mockName('handleChange');
  const amountInput = () => (
    <AmountInput
      id="amount"
      label="Amount"
      value={0}
      handleChange={handleChange}
    />
  );

  it('renders correctly', () => {
    const wrapper = mount(amountInput());
    expect(wrapper).toMatchSnapshot();
  });

  it('handles change as number', () => {
    const wrapper = shallow(amountInput());
    wrapper.find('#amount').simulate('change', { target: { value: '42' } });
    expect(handleChange).toHaveBeenCalledWith(42);
  });
});
