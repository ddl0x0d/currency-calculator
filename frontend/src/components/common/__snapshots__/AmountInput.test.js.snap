// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmountInput /> renders correctly 1`] = `
<AmountInput
  handleChange={[MockFunction handleChange]}
  id="amount"
  label="Amount"
  step={1}
  value={0}
>
  <FormGroup
    as="div"
  >
    <div
      className="form-group"
    >
      <FormLabel
        className="small mb-0"
        column={false}
        srOnly={false}
      >
        <label
          className="small mb-0 form-label"
        >
          Amount
        </label>
      </FormLabel>
      <FormControl
        as="input"
        id="amount"
        min={0}
        onChange={[Function]}
        size="sm"
        step={1}
        type="number"
        value={0}
      >
        <input
          className="form-control form-control-sm"
          id="amount"
          min={0}
          onChange={[Function]}
          step={1}
          type="number"
          value={0}
        />
      </FormControl>
    </div>
  </FormGroup>
</AmountInput>
`;
